packages/components/src/components/custom-select/custom-select.lite.tsx

```diff
@@ -368,17 +372,23 @@ export default function DBCustomSelect(props: DBCustomSelectProps) {
 				state.handleArrowDownUp(event);
 			}
 		},
-		handleClose: (event: any) => {
+		handleClose: (
+			event?: InteractionEvent<HTMLDetailsElement> | void,
+			forceClose?: boolean
+		) => {
 			if (detailsRef) {
-				if (event === 'close') {
+				if (forceClose) {
 					detailsRef.open = false;
 					state.handleSummaryFocus();
-				} else if (detailsRef.open && event?.relatedTarget) {
-					const relatedTarget = event.relatedTarget as HTMLElement;
-					if (!detailsRef.contains(relatedTarget)) {
-						// We need to use delay here because the combination of `contains`
-						// and changing the DOM element causes a race condition inside browser
-						delay(() => (detailsRef.open = false), 1);
+				} else if (detailsRef.open && event) {
+					if (event.relatedTarget) {
+						const relatedTarget =
+							event.relatedTarget as HTMLElement;
+						if (!detailsRef.contains(relatedTarget)) {
```

Comment:
Also close if focus got somehow completely lost.

```suggestion
						if (!event.relatedTarget || !detailsRef.contains(relatedTarget as Node)) {
```